### ISSUE TITLE
[BUG] Update inventory for not existing SKU by USING FEED POST_INVENTORY_AVAILABILITY_DATA

### ISSUE THREAD

**Issue by abuzuhri** [amzn/selling-partner-api-docs#1910]
I try to update inventory by using feed for existing SKU and its working perfect 100% 

The problem when I try to Update inventory for not existing SKU, here XML feed.
````xml
<?xml version="1.0" encoding="utf-16"?>
<AmazonEnvelope xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <Header>
    <DocumentVersion>1.02</DocumentVersion>
    <MerchantIdentifier>A3J37AJU4O9RHK</MerchantIdentifier>
  </Header>
  <MessageType>Inventory</MessageType>
  <Message>
    <MessageID>1</MessageID>
    <OperationType>Update</OperationType>
    <Inventory>
      <SKU>82010312061.22...</SKU>
      <Quantity>2</Quantity>
      <FulfillmentLatency>11</FulfillmentLatency>
    </Inventory>
  </Message>
</AmazonEnvelope>
````
create doc we call (POST /feeds/2021-06-30/documents)
Then CreateFeed and here result 
{x-amzn-RequestId=79d051ca-39af-4558-b0da-a06bdfbc2c6b}
{X-Amzn-Trace-Id=Root=1-61e1784d-0bf0c3a509327d725d964c9e}
response
{"feedId":"232406019006"}
ResultFeedDocumentId=amzn1.tortuga.3.0a3ffe93-ff1e-4a99-baaa-707a0fa21b1f.T1ZKAL0SL02MHS

The final result after get get DocumentProcessingReport show number of message MessagesProcessed=1 and MessagesSuccessful=1 and MessagesWithError=0

and that not true its should show MessagesProcessed=1 and MessagesSuccessful=0 and MessagesWithError=1 and aslo should show the error message

Another example when i try to change price by using feed submit for not existing SKU , but in this case its showing as its should and return the result we need  MessagesProcessed=1 and MessagesSuccessful=0 and MessagesWithError=1

here example XML for change price 
````XML
<?xml version="1.0" encoding="utf-16"?>
<AmazonEnvelope xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <Header>
    <DocumentVersion>1.02</DocumentVersion>
    <MerchantIdentifier>A3J37AJU4O9RHK</MerchantIdentifier>
  </Header>
  <MessageType>Price</MessageType>
  <Message>
    <MessageID>1</MessageID>
    <OperationType>Update</OperationType>
    <Price>
      <SKU>82010312061.22...</SKU>
      <StandardPrice currency="AED">201.05</StandardPrice>
    </Price>
  </Message>
</AmazonEnvelope>
````
create doc we call (POST /feeds/2021-06-30/documents)
Then CreateFeed and here result 
{x-amzn-RequestId=68422343-730b-4135-b8e9-03a4b7d87a08}
{X-Amzn-Trace-Id=Root=1-61e17626-143491b11d6d4bb5030b61f3}
response
{"feedId":"232404019006"}
ResultFeedDocumentId=amzn1.tortuga.3.7642fa60-e9a9-41f8-a5a2-a502999a2841.T2BTUWLSGLTM3Q

ResultDescription
`This SKU is not in the Amazon catalog. If you are receiving this message after submitting a multi-marketplace inventory file and the designated marketplace for this error is different than the marketplace in which you submitted your file, this error is an indication that the Detail page for this item does not exist in the designated marketplace. Amazon is attempting to create the Detail Page for this item on your behalf. If successful, your listing will be created in the designated marketplace within 48 hours.`

So its look like the problem is internal bug inside Amazon Sp API because we submit same SKU for same sellerID and same DeveloperID , in the first case when we try to change Quantity not work as it should but when we try to change the price it's work as expected.

**Note : the problem only when i try to update Quantity  for not exist SKU but using Feed**

**Comment by chapmanjw** [comment]
@abuzuhri The best course here is to open a support case so that your specific submissions can be investigated.

**Comment by abuzuhri** [comment]
> @abuzuhri The best course here is to open a support case so that your specific submissions can be investigated. 

Case open as well #7581285672

But I think it's General problem not related to my only

**Comment by abuzuhri** [comment]
Hi @chapmanjw 

I write all details for the case but I get this answer from amazon , its very strange answer not related bug at all

```
We understand that you are having issues with your feed upload.

Being that the file has been created from a third-party feed, we cannot troubleshoot the issues that are being experienced.

Please provide us a valid New Batch ID which consist of detailed information for your uploads.

To find the Batch ID of your recent file upload please follow the below steps:

1. Go to Seller Central > Inventory
2. Under Inventory tab, click on 'Add Products via Upload'
3. Click on 'Monitor Upload Status' tab
4. You will see a table with 'Batch ID' as second column
```

**Comment by chapmanjw** [comment]
@abuzuhri I will follow up on that case.

**Comment by abuzuhri** [comment]
> @abuzuhri I will follow up on that case.

Thanks @chapmanjw

**Comment by abuzuhri** [comment]
Hello @chapmanjw 

Any Update on this case ?

**Comment by abuzuhri** [comment]
Hello @chapmanjw

Any progress here ?

**Comment by jamieroyce** [comment]
I've been having this exact same issue recently. I've narrowed it down to when I am uploading a feed that contains a SKU that does not exist in Amazon the entire feed seems to error, yet I get no error message back and the feed appears to have successfully processed. I resolved this temporarily by limiting the size of the feed into smaller batches as at the time of processing, I don't know what SKUs are valid and what are not. This does look like an Amazon bug and it should be fixed to return a proper error message, or at least update the SKUs that do exist.
